DataFormats/SoATemplate/interface/SoALayout.h

```diff
@@ -477,60 +538,651 @@
               BOOST_PP_EXPAND(_STREAMER_READ_SOA_DATA_MEMBER_IMPL TYPE_NAME))
+
+// clang-format off
+#define _DECLARE_SOA_DATA_MEMBER_IMPL(VALUE_TYPE, CPP_TYPE, NAME, ARGS) \
+_SWITCH_ON_TYPE(VALUE_TYPE,                                                                                            \
+  /* Scalar */                                                                                                         \
+  CPP_TYPE* BOOST_PP_CAT(NAME, _) EDM_REFLEX_SIZE(scalar_) = nullptr;                                                  \
+  ,                                                                                                                    \
+  /* Column */                                                                                                         \
+  CPP_TYPE * BOOST_PP_CAT(NAME, _) EDM_REFLEX_SIZE(elements_) = nullptr;                                               \
+  ,                                                                                                                    \
+  /* Eigen column */                                                                                                   \
+  size_type BOOST_PP_CAT(NAME, ElementsWithPadding_) = 0; /* For ROOT serialization */                                 \
+  CPP_TYPE::Scalar * BOOST_PP_CAT(NAME, _) EDM_REFLEX_SIZE(BOOST_PP_CAT(NAME, ElementsWithPadding_)) = nullptr;        \
+  byte_size_type BOOST_PP_CAT(NAME, Stride_) = 0;                                                                      \
+)
+// clang-format on
+
+#define _DECLARE_SOA_DATA_MEMBER(R, DATA, TYPE_NAME)                                        \
+  BOOST_PP_IF(BOOST_PP_GREATER(BOOST_PP_TUPLE_ELEM(0, TYPE_NAME), _VALUE_LAST_COLUMN_TYPE), \
+              BOOST_PP_EMPTY(),                                                             \
+              BOOST_PP_EXPAND(_DECLARE_SOA_DATA_MEMBER_IMPL TYPE_NAME))
+
+// clang-format off
+#define _COPY_VIEW_COLUMNS_IMPL(VALUE_TYPE, CPP_TYPE, NAME, ARGS)           \
+  _SWITCH_ON_TYPE(VALUE_TYPE, /* Scalar */                                  \
+                  memcpy(BOOST_PP_CAT(this->metadata().addressOf_, NAME)(), \
+                         BOOST_PP_CAT(view.metadata().addressOf_, NAME)(),  \
+                         sizeof(CPP_TYPE));                                 \
+                  , /* Column */                                            \
+                  memcpy(BOOST_PP_CAT(this->metadata().addressOf_, NAME)(), \
+                         BOOST_PP_CAT(view.metadata().addressOf_, NAME)(),  \
+                         view.metadata().size() * sizeof(CPP_TYPE));        \
+                  , /* Eigen column */                                      \
+                  memcpy(BOOST_PP_CAT(this->metadata().addressOf_, NAME)(), \
+                         BOOST_PP_CAT(view.metadata().addressOf_, NAME)(),  \
+                         BOOST_PP_CAT(NAME, ElementsWithPadding_) * sizeof(CPP_TYPE::Scalar));)
```

Comment:
You simplify this because you require that the individual columns are consecutive in both the source and destination ?
If so can you point me to the check ?